### PR TITLE
Adds a Plushie Delivery Beacon to the game.

### PR DESCRIPTION
**Pull request by ClownMoff**
## About The Pull Request

Adds a Plushie Delivery Beacon

## Why It's Good For The Game

"Oh man I wish I had a moth plushie right now" is no longer a problem. With this NEW system you can INMEDIATLY call a plushie to any location! The donator plushies are out of the pool. This PR asumes this PR is already merged https://github.com/BeeStation/BeeStation-Hornet/pull/10387 as it includes the plushies inside. 

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

imagen
imagen
imagen
imagen
imagen
imagen
imagen
imagen
imagen

imagen

</details>

## Changelog
:cl: Varo (with the help of the good people in discord)
add: Adds a plushie delivery beacon. 
add: Adds a plushie lover trait (+1)
tweak: changes the instrument delivery beacon sprite
tweak: changes the "musician" trait description to be more representative of its actual use. 
/:cl:

### CODE REVIEW

**Comment by Rukofamicom** [comment]
The instrument delivery beacon is tied to a trait that players select and should not be added to a vending machine.

Plushie beacon would be a good choice for such a trait as well instead of available directly from a vending machine.

Plushies (random assortments or otherwise) are one of the things I think would be best suited to a cargo crate order - someone you work to earn money for and could be aiming to collect, especially with the associated trait I am suggesting.

File: code/game/objects/items/plushes.dm [hunk 1]
```diff
@@ -691,11 +716,95 @@
...
+	icon_state = "gangtool-plushie"
+	var/static/list/plushie_list
+
+/obj/item/choice_beacon/radial/plushie/Initialize(mapload)
+	. = ..()
+	plushie_list = list(
+		/obj/item/toy/plush/bubbleplush,
+		/obj/item/toy/plush/carpplushie,
+		/obj/item/toy/plush/plushvar,
+		/obj/item/toy/plush/narplush,
```
Comment (Tyranicranger4) [inline code/game/objects/items/plushes.dm]:
```suggestion
		/obj/item/toy/plush/narplush/hugbox,
```

Non Hugbox narplushies have a special interaction with Cult runes which allows them to act as an Invoker, up to twice per plush. You probably shouldn't be able to get them off spawn from a quirk.

File: code/game/objects/items/plushes.dm [hunk 2]
```diff
@@ -696,6 +721,85 @@
...
+		/obj/item/toy/plush/slimeplushie,
+		/obj/item/toy/plush/nukeplushie,
+		/obj/item/toy/plush/awakenedplushie,
+		/obj/item/toy/plush/beeplushie,
+		/obj/item/toy/plush/crossed,
+		/obj/item/toy/plush/rouny,
+		/obj/item/toy/plush/runtime,
+		/obj/item/toy/plush/flushed,
+		/obj/item/toy/plush/gondola,
+		/obj/item/toy/plush/spessgondolaplush,
```
Comment (Dejaku51) [inline code/game/objects/items/plushes.dm]: doesn't exist
```suggestion
```

File: code/game/objects/items/plushes.dm [hunk 3]
```diff
@@ -696,6 +721,84 @@
...
+		/obj/item/toy/plush/moth/rosy,
+		/obj/item/toy/plush/moth/royal,
+		/obj/item/toy/plush/moth/snow,
+		/obj/item/toy/plush/moth/whitefly,
+		/obj/item/toy/plush/moth/witchwing
+	)
+
+/obj/item/choice_beacon/radial/plushie/generate_options(mob/living/M)
+	var/list/item_list = generate_item_list()
+	if(!item_list.len)
```
Comment (Absolucy) [inline code/game/objects/items/plushes.dm]:
```suggestion
	if(!length(item_list))
```
(slightly faster, and won't runtime)

File: code/game/objects/items/plushes.dm [hunk 4]
```diff
@@ -696,6 +721,84 @@
 	desc = "Hgrgrhrhg cute."
 	icon_state = "flushplush"
 
+/obj/item/choice_beacon/radial/plushie
+	name = "plushie delivery beacon"
+	desc = "Summon your new friend!"
+	icon_state = "gangtool-plushie"
+	var/static/list/plushie_list
+
+/obj/item/choice_beacon/radial/plushie/Initialize(mapload)
```
Comment (Absolucy) [inline code/game/objects/items/plushes.dm]: you don't even need to stick this in /Initialize, can just put the = list(...) right next to the static declaration

File: code/game/objects/items/plushes.dm [hunk 5]
```diff
@@ -696,6 +721,84 @@
...
+		/obj/item/toy/plush/moth/rainbow,
+		/obj/item/toy/plush/moth/redish,
+		/obj/item/toy/plush/moth/rosy,
+		/obj/item/toy/plush/moth/royal,
+		/obj/item/toy/plush/moth/snow,
+		/obj/item/toy/plush/moth/whitefly,
+		/obj/item/toy/plush/moth/witchwing
+	)
+
+/obj/item/choice_beacon/radial/plushie/generate_options(mob/living/M)
```
Comment (Absolucy) [inline code/game/objects/items/plushes.dm:766]: single-letter var names hurt :(

File: code/game/objects/items/plushes.dm [hunk 6]
```diff
@@ -696,6 +721,84 @@
...
+					balloon_alert(M, "[uses] use[uses > 1 ? "s" : ""] remaining")
+					to_chat(M, "<span class='notice'>[uses] use[uses > 1 ? "s" : ""] remaining on the [src].</span>")
+				return
+
+/obj/item/choice_beacon/radial/plushie/generate_item_list()
+	var/static/list/item_list
+	if(!item_list)
+		item_list = list()
+		for(var/V in plushie_list)
+			var/obj/item/toy/plush/I = V
```
Comment (Absolucy) [inline code/game/objects/items/plushes.dm]:
```suggestion
		for(var/obj/item/toy/plush/I as() in plushie_list)
```

**Review by Absolucy — CHANGES_REQUESTED** [review]

File: code/game/objects/items/plushes.dm (hunk 5 above)
Comment (Rukofamicom) [reply code/game/objects/items/plushes.dm:766]: This is currently standard practice in the codebase when declaring mobs, for better or worse.

Comment (Absolucy) [reply code/game/objects/items/plushes.dm:766]: doesn't mean we can't do better with new code :)

Comment (Rukofamicom) [reply code/game/objects/items/plushes.dm:766]: I am actively against changing this standard unless it is standardized across the board like it currently is. Naming them all "Mob" instead of "M"

You are free to open a PR that does this if you want to, but breaking a harmless standard only makes the code messier as a whole. This is not an important variable that will cause confusion elsewhere.

**Review by PestoVerde322 — APPROVED** [review]

**Comment by Nakalon** [comment]
You should get a positive mood buff as long as you have the plushie in your inventory similar to Family Heirloom if this is worth 1 point, I think!

File: code/game/objects/items/plushes.dm (hunk 1 above)
Comment (Rukofamicom) [reply code/game/objects/items/plushes.dm]: This request should not be dismissed - the trait should only give the cosmetic variant of the plushie if at all.

File: code/datums/traits/positive_quirk.dm [hunk 7]
```diff
@@ -89,7 +89,7 @@
 
 /datum/quirk/musician
 	name = "Musician"
-	desc = "You can tune handheld musical instruments to play melodies that clear certain negative effects and soothe the soul."
+	desc = "You can tune handheld musical instruments to play melodies that clear certain negative effects and soothe the soul. You start with a delivery beacon."
```
Comment (Rukofamicom) [inline code/datums/traits/positive_quirk.dm:92]: This change is unrelated to the PR and should be reverted.

File: code/modules/vending/games.dm [hunk 8]
```diff
@@ -11,7 +11,8 @@
 		            /obj/item/toy/cards/deck/cas/black = 3,
 		            /obj/item/toy/cards/deck/unum = 3,
 					/obj/item/toy/cards/deck/tarot = 3,
-					/obj/item/hourglass = 2)
```
Comment (Rukofamicom) [inline code/modules/vending/games.dm:14]: This minor change is unrelated to the PR and should be reverted.

Also how does this hanging comma not cause issues 🤔

**Review by Rukofamicom — CHANGES_REQUESTED** [review]

File: code/datums/traits/positive_quirk.dm (hunk 7 above)
Comment (Rukofamicom) [reply code/datums/traits/positive_quirk.dm:92]: Actually ignore this I'm just being picky for no good reason here upon reading my own review over.

File: code/modules/vending/games.dm (hunk 8 above)
Comment (itsmeow) [reply code/modules/vending/games.dm:14]: Hanging commas are standard in lists like this so you don't have to make unnecessary changes to add a new line.

I also don't really mind since it's a small code cleanup and a one line diff.

File: code/datums/traits/positive_quirk.dm [hunk 9]
```diff
@@ -284,3 +284,21 @@
 /datum/quirk/proskater/on_spawn()
 	var/mob/living/carbon/human/H = quirk_target
 	H.equip_to_slot_or_del(new /obj/item/melee/skateboard/pro(H), ITEM_SLOT_BACKPACK)
+
+/datum/quirk/plushielover
+	name = "Plushie lover"
```
Comment (itsmeow) [inline code/datums/traits/positive_quirk.dm]: All quirk names should be Title Capitalized.

```suggestion
	name = "Plushie Lover"
```

**Review by Rukofamicom — APPROVED** [review]
Didn't really have to remove both of the god plushies, but it is definitely safer this way in the long run.
